### ISSUE TITLE
Allow out-of-order event registration

### ISSUE THREAD

**Issue by UkoeHB** [projectharmonia/bevy_replicon#183]
We can do this by storing registered events in a sorted list of typeids, then assigning channel ids based on index.

**Comment by Shatur** [comment]
Agree, but shouldn't use [TypeId](https://doc.rust-lang.org/std/any/struct.TypeId.html) since it's not stable across binaries. [Relevant issue](https://github.com/bevyengine/bevy/issues/32) in Bevy.
But it looks like we can just use type names.

**Comment by Shatur** [comment]
Hm... Yes, I missed it. That just what `@cart` suggested in https://github.com/bevyengine/bevy/issues/32#issuecomment-673784868.
Then I'm not sure how to solve it properly.
